### PR TITLE
Use GResources, GtkBuilder, GActions.

### PR DESCRIPTION
**Pull request by Obsidien**
The IToolbar madness is a quite complicated way to do things… and not only because there’s a GtkToolbar called “Menubar” and and GtkHeaderBar called “Toolbar” (and no GtkMenuBar at all). GActions should be used for that, and I think it’d be better for the UI to be managed by GtkBuilder.

GResources are a great way to have files loaded, that will permit to clean some more code. But it needs quite some CMake hacks, and I’m not good at that. I used some common files from LaTeXila, from the time it was using CMake, but the code I wrote is probably not clean. Please check.

I’d like to discuss the integration with GNOME… The application is not greatly integrated, it should use a standard-high headerbar, symbolic icons, etc. Why are we in this current state of the things ? How can we do better ? Can I add the support of the app-menu ?.. I’d prefer of course not to fork.

### CODE REVIEW

**Comment by voldyman** [comment]
IToolbar method was used because, i needed to add the support quickly (a friend required it) this approach is definetely better.
GResources were a goal but i wasn't getting enough time to read up on it and add it here.
Integration with GNOME would be awesome, i am a elementary OS user so haven't really tried it on gnome.

an issue already exists for symbolic icons which will be incorporated.

We are in the current state of things because i have been travelling too much, my laptop got damaged in the journey and has gone for repair. 

Yes you can add support for app-menu.

no forks please, spoon are good even chopsticks would do. :smile_cat:

**Comment by voldyman** [comment]
@emersion can you review this? i don't have a working laptop right now.

**Comment by emersion** [comment]
@voldyman Okay, I do it right now ;-)

File: src/Widgets/Window.vala [hunk 1]
```diff
@@ -179,23 +197,34 @@ public class Window : Gtk.Window {
...
+            Gtk.Box layout = (Gtk.Box) builder.get_object ("layout");
+
+            layout.pack_start (box, false);
+            add (layout);
         }
 
-        var box = new Gtk.Paned (Gtk.Orientation.HORIZONTAL);
-        box.expand = true;
+        Gtk.IconTheme icon_theme = Gtk.IconTheme.get_default ();
+        Gtk.Image fallback1 = (Gtk.Image) builder.get_object ("fallback1");
```
Comment (emersion) [inline src/Widgets/Window.vala]: Maybe rename this to `menu-icon` or something like this? (And the next one too: `export-icon`)

**Comment by emersion** [comment]
Reviewed, apart from the object name in my previous comment, this looks good to me. :-)

**Comment by emersion** [comment]
LGTM, thanks!

**Comment by voldyman** [comment]
@emersion @Obsidien the build broke

```
XMLLINT not set and xmllint not found in path; skipping xml preprocessing.
```

Detailed build log
https://launchpadlibrarian.net/199659835/buildlog_ubuntu-trusty-amd64.mark-my-words_0.1-0%7E135%7Eubuntu14.04.1_BUILDING.txt.gz

**Comment by emersion** [comment]
`xmllint` is a part of the `libxml2-utils` package on Ubuntu, adding it to dependencies can solve the problem.

**Comment by voldyman** [comment]
@emersion pullrequest? :)

**Comment by emersion** [comment]
See #51, don't know if I need to add it somewhere else.

**Comment by voldyman** [comment]
builds failing due to some other issue

```
make[3]: Entering directory `/build/buildd/mark-my-words-0.1/obj-x86_64-linux-gnu'
/usr/bin/cmake -E cmake_progress_report /build/buildd/mark-my-words-0.1/obj-x86_64-linux-gnu/CMakeFiles 11
[ 36%] Generating src/mark-my-words-resources.c
cd /build/buildd/mark-my-words-0.1/resources && /usr/bin/glib-compile-resources --generate-source --target=/build/buildd/mark-my-words-0.1/obj-x86_64-linux-gnu/src/mark-my-words-resources.c mark-my-words.gresource.xml
can't write to file /build/buildd/mark-my-words-0.1/obj-x86_64-linux-gnu/src/mark-my-words-resources.cmake[3]: *** [src/mark-my-words-resources.c] Error 1
make[3]: Leaving directory `/build/buildd/mark-my-words-0.1/obj-x86_64-linux-gnu'
make[2]: *** [CMakeFiles/mark-my-words.dir/all] Error 2
make[2]: Leaving directory `/build/buildd/mark-my-words-0.1/obj-x86_64-linux-gnu'
make[1]: *** [all] Error 2
make[1]: Leaving directory `/build/buildd/mark-my-words-0.1/obj-x86_64-linux-gnu'
dh_auto_build: make -j1 returned exit code 2
make: *** [build-arch] Error 2
```

detailed bug report https://launchpadlibrarian.net/199665385/buildlog_ubuntu-vivid-amd64.mark-my-words_0.1-0~137~ubuntu15.04.1_BUILDING.txt.gz

**Comment by Obsidien** [comment]
[Looks like](https://developer.gnome.org/gio/stable/gio-GResource.html) the XMLLINT dependency has been added for the GResource to use “xml-stripblank” (see the GResource file), but it normally should just skip if the executable is not found. May be a compilation choice.

**Comment by emersion** [comment]
@Obsidien The error message about `xmllint` was a warning and not the fatal error. Don;t know how to solve this one though.

**Pull request merged**